lib/Pipes/TaggedFunctionKind.cpp

```diff
@@ -49,15 +49,19 @@ TaggedFunctionKind::compactTargets(const pipeline::Context &Ctx,
     if (Target.getPathComponents().back().isAll())
       return pipeline::TargetsList({ AllFunctions });
 
-    Set.insert(Target.getPathComponents().back().getName());
+    TargetsSet.insert(Target.getPathComponents().back().getName());
   }
 
-  // check if all functions in the model are in the targets
+  // check if all functions in the model, that are not fake, are in the targets.
   // if they are, return *
-  if (llvm::all_of(Model.Functions, [&Set](const model::Function &F) {
-        return Set.contains(F.Entry.toString());
-      })) {
-
+  const auto IsNotFake = [](const auto &F) {
+    return F.Type != model::FunctionType::Fake;
+  };
+  const auto IsInTargetSet = [&TargetsSet](const model::Function &F) {
+    return TargetsSet.contains(F.Entry.toString());
+  };
+  if (llvm::all_of(llvm::make_filter_range(Model.Functions, IsNotFake),
+                   IsInTargetSet)) {
     return pipeline::TargetsList({ AllFunctions });
   }
```

Comment:
Why do you need this change?
Was it crashing?

Reply:
I don't remember exactly if it was crashing, but it wasn't working, either crashing or reporting an error.

In particular, whenever I had even a single `Fake` function in the model, the `*` target wasn't working properly. The fake functions would be found in the model but they wouldn't have a counterpart in LLVM IR, because they were not tagged as `Isolated` (which IMHO makes sense, given that you cannot really isolate a fake function). So the `AllFunctions` target would be deemed impossible to produce, yielding a failure.

So, discussing with @drblallo we reached the conclusion that the sane thing to do was not consider `Fake` functions. Do you have a better idea?

Reply:
One day fake functions will disappear. They will just be regular functions with an `inline` attribute.
They will be inlined in callers but we could decompile them individually nevertheless.
For now this makes sense, but it'll change.

Reply:
If we're gonna have fake functions in the LLVM IR as well, that's great! I'll be more than happy to revert this! It also means we're gonna be able to show C code for fake functions, which is not currently possible.

However, for now this is necessary for properly handling binaries with fake functions. Downstream tests in `revng-c` don't pass without this at the moment.